glean/update.go

```diff
@@ -0,0 +1,120 @@
+package glean
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"net/http"
+	"io"
+	"io/ioutil"
+	"strings"
+	"runtime"
+	"log"
+	"path/filepath"
+)
+
+func buildGleanReleaseName() string {
+	var arch string
+	if runtime.GOARCH == "amd64" {
+		arch = "x86_64"
+	} else{
+		arch = "arm64"
+	}
+	
+	switch runtime.GOOS {
+	case "windows":
+		name := fmt.Sprintf("glean_%s_%s.zip", "Windows", arch)
+		return name
+	default:
+		name := fmt.Sprintf("glean_%s_%s.tar.gz", strings.Title(runtime.GOOS), arch)
+		return name
+	}
+}
+
+func getLatestVersion() string {
+	
+	response, err := http.Get(urlBase + "/glean/releases/download/?mirror_intel_list")
+	if err != nil {
+		panic(err.Error())
+	}
+	defer response.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	bodyText := string(body)
+	index := strings.Index(bodyText, "v0")
+	var latestVersion string
+	for index != -1 {
+		latestVersion = bodyText[index:index+6]
+		fmt.Println("Found latest version:", latestVersion)
```

Comment:
newer

Reply:
This function only finds the latest version and doesn't do comparison, so I think latest is better.